lib/cocoapods/installer/xcode/pods_project_generator/target_installer.rb

```diff
@@ -89,10 +89,44 @@ def custom_build_settings
             settings
           end
 
+          # @param [Generator] generator
+          #        the generator to use for generating the content.
+          #
+          # @param [Pathname] path
+          #        the pathname to save the content into.
+          #
+          # Saves the content the provided path unless the path exists and the contents are exactly the same.
+          #
+          # @return [Void]
+          #
+          def update_changed_file(generator, path)
+            if path.exist?
+              generator.save_as(support_files_temp_dir)
+              unless FileUtils.identical?(support_files_temp_dir, path)
+                generator.save_as(path)
+              end
+            else
+              generator.save_as(path)
+            end
+            clean_support_files_temp_dir if support_files_temp_dir.exist?
+          end
+
           # Creates the directory where to store the support files of the target.
           #
           def create_support_files_dir
-            target.support_files_dir.mkdir
+            target.support_files_dir.mkpath
+          end
+
+          # Remove temp file whose store .prefix/config/dummy file.
+          #
+          def clean_support_files_temp_dir
+            support_files_temp_dir.rmtree
+          end
+
+          # @return [String] The temp file path to store temporary files.
+          #
+          def support_files_temp_dir
```

Comment:
it is actually a dir, I renamed the actual dir to be better. Let me know if this good